firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -189,6 +196,7 @@ static FirebaseFirestore newInstance(
     // NOTE: We allow firebaseApp to be null in tests only.
     this.firebaseApp = firebaseApp;
     this.instanceRegistry = instanceRegistry;
+    this.mapEncoders = new HashSet<>(mapEncoders);
```

Comment:
Must have a `HashSet` here, otherwise, in the test util, I am not able to change the content of the set (cannot `clear` and `set` back the mappers on a `collection$immutable Set`)